manifests/service.pp

```diff
@@ -0,0 +1,62 @@
+# == Class: xylem::service
+#
+# Installs Xylem and configures and manages the service.
+#
+# === Parameters
+#
+# [*repo_manage*]
+#   If true, xylem::repo will be used to manage the package repository.
+#
+# [*repo_source*]
+#   Repository source passed to xylem::repo.
+#
+# [*package_ensure*]
+#   The ensure value for the seed-xylem package.
+#
+# [*backend*]
+#   Set the rhumba backend to use for job queue
+#
+# [*redis_host*]
+#   Redis host to use when using the redis Rhumba backend
+#
+# [*redis_port*]
+#   Redis port to use when using the redis Rhumba backend
+#
+class xylem::service (
+  $backend           = undef,
+  $redis_host        = undef,
+  $redis_port        = undef,
+
+  $repo_manage       = true,
+  $repo_source       = 'p16n-seed',
+  $package_ensure    = 'installed',
+){
+  validate_bool($repo_manage)
+
+  class { 'xylem::config':
+    backend    => $backend,
+    redis_host => $redis_host,
+    redis_port => $redis_port,
+  }
+
+  if $repo_manage {
+    class { 'xylem::repo':
+      manage => $repo_manage,
+      source => $repo_source,
+    }
+  }
+
+  package { 'seed-xylem':
```

Comment:
Originally this was `xylem::docker` for the docker client side and `xylem::node` for the backend storage/database side. I'm happy to turn `xylem::service` into plain `xylem` for now, but I the other changes probably belong in their own branch.

That said, this whole thing has turned into another refactoring exercise (like #9) instead of just adding support for the `marathon_sync` plugin, so maybe we should do the rest of the refactoring here and open a new ticket for the actual `marathon_sync` work.